code/datums/gamemode/factions/bloodcult/bloodcult_projectiles.dm

```diff
@@ -0,0 +1,284 @@
+				blade.forceMove(loc)
+				blade.attack_hand(M)
+				blade = null
+				qdel(src)
+		else
+			A.attackby(blade,shade)
+	else
+		if (ismob(A))
+			var/mob/M = A
+			if (!iscultist(M))
+				A.hitby(blade)
+		else
+			A.hitby(blade)
+	if(isliving(A))
+		forceMove(get_step(loc,dir))
+		bump_original_check()
+	else
+		..()
+
+/obj/item/projectile/soulbullet/bump_original_check()
+	if (loc == target && !redirected)
+		redirect()
+
+/obj/item/projectile/soulbullet/reset()
+	..()
+	secondary_target = target
+
+
+/obj/item/projectile/soulbullet/proc/redirect()
+	redirected = 1
+	projectile_speed = 0.66
+	if (target == secondary_target)
+		return
+	starting = target
+	target = secondary_target
+	dist_x = abs(target.x - starting.x)
+	dist_y = abs(target.y - starting.y)
+	override_starting_X = starting.x
+	override_starting_Y = starting.y
+	override_target_X = target.x
+	override_target_Y = target.y
+	if (target.x > starting.x)
+		dx = EAST
+	else
+		dx = WEST
+	if (target.y > starting.y)
+		dy = NORTH
+	else
+		dy = SOUTH
+	if(dist_x > dist_y)
+		error = dist_x/2 - dist_y
+	else
+		error = dist_y/2 - dist_x
+
+/obj/item/projectile/soulbullet/cultify()
+	return
+
+
+/obj/item/projectile/soulbullet/Cross(var/atom/movable/mover, var/turf/target, var/height=1.5, var/air_group = 0)
+	if(istype(mover, /obj/item/projectile))
+		if (prob(30))//less likely to be hit when perforating
+			return 0
+	return ..()
+
+/obj/item/projectile/soulbullet/attackby(var/obj/item/I, var/mob/user)
+	if (blade)
+		return blade.attackby(I,user)
+
+/obj/item/projectile/soulbullet/hitby(var/atom/movable/AM)
+	if (blade)
+		return blade.hitby(AM)
+
+/obj/item/projectile/soulbullet/bullet_act(var/obj/item/projectile/P)
+	if (blade)
+		return blade.bullet_act(P)
+
+//////////////////////////////
+//                          //
+//        BLOOD SLASH       ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//                          //Used when a cultist swings a soul blade that has at least 5 blood in it.
+//////////////////////////////
+
+/obj/item/projectile/bloodslash
+	name = "soul blade"
+	icon = 'icons/obj/projectiles_experimental.dmi'
+	icon_state = "bloodslash"
+	damage = 15
+	damage_type = BURN
+	flag = "energy"
+	custom_impact = 1
+
+/obj/item/projectile/bloodslash/Destroy()
+	var/turf/T = get_turf(src)
+	playsound(T, 'sound/effects/forge_over.ogg', 100, 1)
+	if (!locate(/obj/effect/decal/cleanable/blood/splatter) in T)
+		var/obj/effect/decal/cleanable/blood/splatter/S = new (T)//splash
+		S.amount = 1
+	..()
+
+/obj/item/projectile/bloodslash/to_bump(var/atom/A)
+	if (isliving(A))
+		forceMove(A.loc)
+		var/mob/living/M = A
+		if (!iscultist(M))
+			..()
+	qdel(src)
+
+/obj/item/projectile/bloodslash/on_hit(var/atom/target, var/blocked = 0)
+	if (isliving(target))
+		var/mob/living/M = target
+		if(M.flags & INVULNERABLE)
+			return 0
+		if (iscultist(M))
+			return 0
+		if (M.stat == DEAD)
```

Comment:
I think you're not supposed to check stat directly and use `isDead` (for lings for example)